data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -617,6 +619,40 @@ void getValidatorInclusionAtEpoch_shouldThrowIllegalArgForCurrentOrFutureEpoch()
         .isInstanceOf(IllegalArgumentException.class);
   }
 
+  @Test
+  void checkMinimumMilestone_throwsCorrectMessage() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(spec, recentChainData, combinedChainDataClient, rewardCalculatorMock);
+    final Optional<StateAndMetaData> maybeState =
+        Optional.of(
+            new StateAndMetaData(
+                data.randomBeaconState(), SpecMilestone.PHASE0, false, true, true));
+    assertThatThrownBy(
+            () -> provider.checkMinimumMilestone(maybeState, SpecMilestone.ELECTRA, "badbeef"))
+        .isInstanceOf(BadRequestException.class)
+        .hasMessageContaining("badbeef");
+  }
+
+  @Test
+  void checkMinimumMilestone_checksMetadataMilestone() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(spec, recentChainData, combinedChainDataClient, rewardCalculatorMock);
+    final Optional<StateAndMetaData> maybeState =
+        Optional.of(
+            new StateAndMetaData(
+                data.randomBeaconState(), SpecMilestone.ELECTRA, false, true, true));
+    assertDoesNotThrow(
+        () -> provider.checkMinimumMilestone(maybeState, SpecMilestone.ELECTRA, "badbeef"));
+  }
+
+  @Test
+  void checkMinimumMilestone_ignoresEmptyObjects() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(spec, recentChainData, combinedChainDataClient, rewardCalculatorMock);
+    assertDoesNotThrow(
+        () -> provider.checkMinimumMilestone(Optional.empty(), SpecMilestone.ELECTRA, "badbeef"));
+  }
+
```

Comment:
Are we missing a test where we return the list of pending consolidations?

Reply:
thats covered in the unit tests for the endpoint, but we could build that out - none of the `pending_` were tested as its just a field on the state thats returned...

Reply:
The testing infra is kinda lacking for that a little... want me to raise a ticket?

Reply:
added #9328